bcda/auth/client/oktaclient.go

```diff
@@ -31,94 +42,210 @@ func init() {
+	err = json.NewDecoder(resp.Body).Decode(&result)
+	if err != nil {
+		return "", "", err
+	}
 
-func DeleteUser(userId string) (bool, error) {
-	reqId := uuid.NewRandom()
-	client := NewOktaClient()
-	userField := logrus.Fields{"user_id": userId}
+	clientID := result["client_id"].(string)
+	clientSecret := result["client_secret"].(string)
 
-	logRequest(reqId).WithFields(userField).Info("Okta delete user request")
-	resp, err := client.User.DeactivateUser(userId, nil)
+	err = addClientToPolicy(clientID, requestID)
 	if err != nil {
-		logError(err, reqId).WithFields(userField).Error("Okta delete user error")
-		return false, err
+		logError(err, requestID).Infof("invalid client %s", localID)
+		return "", "", err
+		// client will not be able to use server until it is added to the policy
 	}
 
-	logResponse(resp.StatusCode, reqId).WithFields(userField).Info("Okta delete user success")
-	return true, err
+	return clientID, clientSecret, nil
 }
 
-func FindUser(email string) (string, error) {
-	reqId := uuid.NewRandom()
-	client := NewOktaClient()
+type Policy struct {
+	ID          string `json:"id"`
+	Type        string `json:"type"`
+	Status      string `json:"status"`
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Priority    int    `json:"priority"`
+	System      bool   `json:"system"`
+	Conditions  Cond   `json:"conditions"`
+}
+
+type Cond struct {
+	Clients Cli `json:"clients"`
+}
 
-	logRequest(reqId).Info("Okta find request")
-	filter := query.NewQueryParams(query.WithQ(email))
-	users, resp, err := client.User.ListUsers(filter)
+type Cli struct {
+	Include []string `json:"include"`
+}
+
+// Update the Auth Server's access policy to include our new client application. Otherwise, that application
+// will not be able to use the server. To do this, we first get the current list of clients, add the new
+// server to the inclusion list, and put it back to the server
+func addClientToPolicy(clientID string, requestID uuid.UUID) error {
+	policyUrl := fmt.Sprintf("%s/api/v1/authorizationServers/%s/policies", oktaBaseUrl, oktaServerID)
+	req, err := http.NewRequest("GET", policyUrl, nil)
 	if err != nil {
-		logError(err, reqId).Error("Okta find error")
-		return "", err
-	}
-
-	l := logResponse(resp.StatusCode, reqId)
-	switch len(users) {
-	case 0:
-		l.Info("Okta user find request unsuccessful")
-		return "", err
-	case 1:
-		userId := users[0].Id
-		userField := logrus.Fields{"user_id": userId}
-		l.WithFields(userField).Info("Okta user find request successful")
-		return userId, err
-	default:
-		l.Info("Okta user find request returned more than one user")
-		return "", err
+		logError(err, requestID)
+		return err
 	}
+
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Accept", "application/json")
+	req.Header.Add("Authorization", oktaAuthString)
+
+	var client = &http.Client{Timeout: time.Second * 10,}
+	resp, err := client.Do(req)
+	if err != nil {
+		logError(err, requestID)
```

Comment:
Mea culpa: I named the log helper methods in a way that promotes misunderstanding.  This line (and all the other logging lines in this function) will give you a *logrus.Entry object.  Unless you assign it (`l := logError(err, requestID)`) or chain it into a method (`logError(err, requestID).Error("oh no!")`) it's not going to do anything.

Reply:
How about if we rename these methods "loggerWithError()" and so forth? That seems like a clearer expression of purpose.

Reply:
That name works for me.

Reply:
`errorLogger()`? 🤷‍♂️